lte/gateway/c/core/oai/test/amf/amf_app_test_util.cpp

```diff
@@ -164,6 +160,217 @@ int send_uplink_nas_registration_complete(
+
+  return rc;
+}
+
+void create_pdu_resource_setup_response_itti(
+    itti_ngap_pdusessionresource_setup_rsp_t* response,
+    amf_ue_ngap_id_t ue_id) {
+  if (!response) return;
+  response->amf_ue_ngap_id                                       = ue_id;
+  response->gnb_ue_ngap_id                                       = 1;
+  response->pduSessionResource_setup_list.item[0].Pdu_Session_ID = 1;
+  response->pduSessionResource_setup_list.no_of_items            = 1;
+  response_gtp_tunnel_t* tunnel =
+      &response->pduSessionResource_setup_list.item[0]
+           .PDU_Session_Resource_Setup_Response_Transfer.tunnel;
+  tunnel->gTP_TEID[0] = 0x0;
+  tunnel->gTP_TEID[1] = 0x0;
+  tunnel->gTP_TEID[2] = 0x0;
+  tunnel->gTP_TEID[3] = 0x1;
+
+  tunnel->transportLayerAddress[0] = 0xc0;
+  tunnel->transportLayerAddress[1] = 0xa8;
+  tunnel->transportLayerAddress[2] = 0x3c;
+  tunnel->transportLayerAddress[3] = 0x96;
+
+  AssociatedQosFlowList_t* qosFlow =
+      &response->pduSessionResource_setup_list.item[0]
+           .PDU_Session_Resource_Setup_Response_Transfer.associatedQosFlowList;
+  qosFlow->items                = 1;
+  qosFlow->QosFlowIdentifier[0] = 9;
+}
+int send_pdu_resource_setup_response(amf_ue_ngap_id_t ue_id) {
+  int rc                                            = RETURNok;
+  itti_ngap_pdusessionresource_setup_rsp_t response = {};
+  create_pdu_resource_setup_response_itti(&response, ue_id);
+
+  amf_app_handle_resource_setup_response(response);
+
+  return rc;
+}
+
+void create_pdu_notification_response_itti(
+    itti_n11_received_notification_t* response) {
+  if (!response) return;
+  std::string imsi = "222456000000001";
+  std::copy(imsi.begin(), imsi.end(), std::begin(response->imsi));
+  response->sm_session_fsm_state = CREATING_0;
+  response->sm_session_version   = 0;
+  response->pdu_session_id       = 1;
+  response->request_type         = EXISTING_PDU_SESSION;
+  response->m5g_sm_capability.multi_homed_ipv6_pdu_session = false;
+  response->m5g_sm_capability.reflective_qos               = false;
+  response->m5gsm_cause      = M5GSM_OPERATION_SUCCESS;
+  response->pdu_session_type = IPV4;
+  response->notify_ue_evnt   = PDU_SESSION_STATE_NOTIFY;
+}
+
+int send_pdu_notification_response() {
+  int rc                                    = RETURNerror;
+  itti_n11_received_notification_t response = {};
+  create_pdu_notification_response_itti(&response);
+
+  rc = amf_app_handle_notification_received(&response);
+
+  return rc;
+}
+
+/* Create pdu session release  request from ue */
+int send_uplink_nas_pdu_session_release_request(
+    amf_app_desc_t* amf_app_desc_p, amf_ue_ngap_id_t ue_id, const plmn_t& plmn,
+    const uint8_t* nas_msg, uint8_t nas_msg_length) {
+  bstring pdu_session_est_req;
+  tai_t originating_tai = {};
+
+  if ((!amf_app_desc_p) || (!nas_msg) || (nas_msg_length == 0)) {
+    return RETURNerror;
+  }
+
+  pdu_session_est_req = blk2bstr(nas_msg, nas_msg_length);
```

Comment:
This is not for establishment, so use a generic variable name
```suggestion
  pdu_session_req = blk2bstr(nas_msg, nas_msg_length);
```

Reply:
addressed review comment